contracts/LinniaRecords.sol

```diff
@@ -99,6 +102,33 @@ contract LinniaRecords is Ownable, Pausable, Destructible {
         return true;
     }
 
+    /// Add a record by user without any provider's signatures and get a reward.
+    ///
+    /// @param dataHash the hash of the data
+    /// @param metadata plaintext metadata for the record
+    /// @param dataUri the data uri path of the encrypted data
+    /// @param token the ERC20 token address for the rewarding token
+    function addRecordwithReward (
+        bytes32 dataHash, string metadata, string dataUri, address token)
```

Comment:
Why is the token an argument? Are we expected to keep balances in many ERC20 tokens?